reference/generators/cmake.rst

```diff
@@ -1,190 +1,220 @@
+  | CONAN_CXX_FLAGS                | Aggregated CXX flags                                                 |
+  +--------------------------------+----------------------------------------------------------------------+
+  | CONAN_SHARED_LINK_FLAGS        | Aggregated Shared link flags                                         |
+  +--------------------------------+----------------------------------------------------------------------+
+  | CONAN_C_FLAGS                  | Aggregated C flags                                                   |
+  +--------------------------------+----------------------------------------------------------------------+
+
+- **User information declared variables**:
+
+  If any of the requirements is filling the :ref:`user_info<method_package_info_user_info>` object in the
+  :ref:`package_info<method_package_info>` method a set of variables will be declared following this naming:
+
+  +--------------------------------+----------------------------------------------------------------------+
+  | NAME                           | VALUE                                                                |
+  +================================+======================================================================+
+  | CONAN_USER_XXXX_YYYY           | User declared value                                                  |
+  +--------------------------------+----------------------------------------------------------------------+
+
+  Where ``XXXX`` means the name of the requirement in uppercase and ``YYYY`` the variable name. For example, if this recipe declares:
+
+  .. code-block:: python
+
+      class MyLibConan(ConanFile):
+          name = "MyLib"
+          version = "1.6.0"
+
+          # ...
+
+          def package_info(self):
+              self.user_info.var1 = 2
+
+  Other library requiring ``MyLib`` and using this generator will get:
+
+  .. code-block:: cmake
+     :caption: *conanbuildinfo.cmake*
+
+      # ...
+      set(CONAN_USER_MYLIB_var1 "2")
+
+.. _conanbuildinfocmake_macros:
+
+Macros available in *conanbuildinfo.cmake*
+------------------------------------------
+
+conan_basic_setup()
++++++++++++++++++++
+
+This is a helper and general purpose macro that uses all the macros below to set all the CMake variables according to the Conan generated
+variables. See the macros below for detailed information.
+
+.. code-block:: cmake
+
+    macro(conan_basic_setup)
+        set(options TARGETS NO_OUTPUT_DIRS SKIP_RPATH KEEP_RPATHS SKIP_STD SKIP_FPIC)
+
+Parameters:
+    - ``TARGETS`` (Optional): Setup all the CMake variables by target (only CMake > 3.1.2). Activates the call to the macro
+      ``conan_target_link_libraries()``.
+    - ``NO_OUTPUT_DIRS`` (Optional): Do not adjust the output directories. Deactivates the call to the macro ``conan_output_dirs_setup()``.
+    - ``SKIP_RPATH`` (Optional): **[DEPRECATED]** Use ``KEEP_RPATHS`` instead. Activate ``CMAKE_SKIP_RPATH`` variable in OSX.
```

Comment:
If it is deprecated, do not document the functionality here

```suggestion
    - ``SKIP_RPATH`` (Optional): **[DEPRECATED]** Use ``KEEP_RPATHS`` instead.
```

Reply:
There were comments regarding deprecation in other issues pointing that we should keep the description. See for example **source_dir** and **build_dir** in here: https://docs.conan.io/en/latest/reference/build_helpers/cmake.html#configure

Reply:
I find it very confusing: _"skip: use keep instead. Activate SKIP"_, and in the next line: _"KEEP: do not adjust SKIP "_ 💥 . But it's just my preference.